boards/common/particle-mesh/Makefile.include

```diff
@@ -2,10 +2,6 @@
 BOARD_NRFANTENNA_DEFAULT ?= BUILTIN
 CFLAGS += -DBOARD_NRFANTENNA_DEFAULT=BOARD_NRFANTENNA_$(BOARD_NRFANTENNA_DEFAULT)
 
-# set default port depending on operating system
-PORT_LINUX ?= /dev/ttyUSB0
-PORT_DARWIN ?= $(firstword $(sort $(wildcard /dev/tty.SLAB_USBtoUART*)))
-
 # add the common header files to the include path
 INCLUDES += -I$(RIOTBOARD)/common/particle-mesh/include
 
```

Comment:
I noticed `PARTICLE_MONOFIRMWARE` is not set by default - how about adding

```make
ifeq (,$(PROGRAMMER) $(DEBUG_ADAPTER))
  PARTICLE_MONOFIRMWARE ?= 1
endif
```

Reply:
I don't know if that's a somehow more sensible default. It certainly is for people who just got their board, but it's a change for anyone who has already flashed their board by any other method (especially given I haven't found the docs on how to get the Particle bootloader back once you've switched).

Reply:
https://github.com/particle-iot/device-os/releases has binaries for the bootloader, can you flash these?

Reply:
IIRC, `Makefile.include` is handled after `Makefile.dep` now. So if PARTICLE_MONOFIRMWARE is set to 1 here but is checked in Makefile.dep like you suggested in https://github.com/RIOT-OS/RIOT/pull/15394#discussion_r518777729, that might not work.

Reply:
I don't have a suitable cable; I was just browsing for one earlier today with little success, so I might just go for the full particle debugger. (It's sad but it seams it's easier to get a few debuggers each with their own cable than use the debuggers I have with their 20pin header and just get an adapter cable to 10pin 50mil).

Reply:
@maribu had a nice idea for the AVR boards, you can do

```make
# PROGRAMMER defaults to dfu-util which uses the particle bootloader
PROGRAMMER_PARTICLE_BORON ?= dfu-util
PROGRAMMER ?= $(PROGRAMMER_PARTICLE_BORON)
```

that way you can set `PROGRAMMER_PARTICLE_BORON` to something else in your `.bashrc` if you removed the bootloader, but users who just got the board don't have to do any additional configuration.

Reply:
At any rate, unless we're making the default stdio conditional on the programmer, picking the default programmer is out of scope here IMO. (I'm happy to do a follow-up for #12751 that makes this the default, though -- I'm just ordering the cable that'll allow me to experiment with different bootloaders on these boards).